src/Unused/CLI/SearchWithProgress.hs

```diff
@@ -1,40 +1,45 @@
+{-# LANGUAGE FlexibleContexts #-}
+
 module Unused.CLI.SearchWithProgress
     ( searchWithProgressBar
     ) where
 
 import Control.Monad.State
+import Data.Bifunctor (first)
 import System.ProgressBar
 import Unused.TermSearch (search)
 
+newtype Progress a = Progress a
+
+instance Functor Progress where
+    fmap f (Progress s) = Progress $ f s
+
+type ProgressState = Progress (Integer, Integer)
+
 searchWithProgressBar :: [String] -> IO [String]
 searchWithProgressBar terms =
-    (concat . fst) <$> runStateT (performSearch $ length terms) terms
-
-performSearch :: Int -> StateT [String] IO [[String]]
-performSearch total = do
-    currentTerm <- getSearchTerm
-    searchResults <- liftIO $ search currentTerm
+    concat . fst <$> runStateT (mapM performSearch terms) progress
```

Comment:
then instead of `performSearch` you could have `perform = performSearch <* incrementCount <* printProgressBar`